tests/test_object.h

```diff
@@ -87,6 +187,119 @@ TEST_CASE("[Object] Metadata") {
 			meta_list2.size() == 0,
 			"The metadata list should contain 0 items after removing all metadata items.");
 }
+
+TEST_CASE("[Object] Construction") {
+	Object object;
+
+	CHECK_MESSAGE(
+			!object.is_reference(),
+			"Object is not a Reference.");
+
+	Object *p_db = ObjectDB::get_instance(object.get_instance_id());
+	CHECK_MESSAGE(
+			p_db == &object,
+			"The database pointer returned by the object id should reference same object.");
+}
+
+TEST_CASE("[Object] Script instance property setter") {
+	Object object;
+	_MockScriptInstance *script_instance = memnew(_MockScriptInstance);
+	object.set_script_instance(script_instance);
+
+	bool valid = false;
+	object.set("some_name", 100, &valid);
+	CHECK(valid);
+	Variant actual_value;
+	CHECK_MESSAGE(
+			script_instance->get("some_name", actual_value),
+			"The assigned script instance should successfully retrieve value by name.");
+	CHECK_MESSAGE(
+			actual_value == Variant(100),
+			"The returned value should equal the one which was set by the object.");
+}
+
+TEST_CASE("[Object] Script instance property getter") {
+	Object object;
+	_MockScriptInstance *script_instance = memnew(_MockScriptInstance);
+	script_instance->set("some_name", 100); // Make sure script instance has the property
+	object.set_script_instance(script_instance);
+
+	bool valid = false;
+	const Variant &actual_value = object.get("some_name", &valid);
+	CHECK(valid);
+	CHECK_MESSAGE(
+			actual_value == Variant(100),
+			"The returned value should equal the one which was set by the script instance.");
+}
+
+TEST_CASE("[Object] Built-in property setter") {
+	ClassDB::register_class<_TestDerivedObject>();
```

Comment:
Just wanted to note that `ClassDB` is used globally throughout tests, but as long as it doesn't affect other tests, this should be fine I think.